sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExternalSource.scala

```diff
@@ -232,4 +231,16 @@ object ExternalSource extends Logging {
       throw new IllegalStateException("No GPU conversion")
     }
   }
+
+  def tagForGpu(
+    ignore: AppendDataExec,
+    meta: AppendDataExecMeta): Unit = {
+    meta.willNotWorkOnGpu(s"AppendDataExec is not supported")
```

Comment:
We can have a test to verify this default behavior, that is "fallback to CPU"

Reply:
Seems need a concreate sub-exec to test fallback.